src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php

```diff
@@ -48,38 +49,44 @@
                 abstract_arg('argument value resolvers'),
             ])
 
+        ->set('argument_resolver.specified_resolver', SpecifiedResolverValueResolver::class)
+            ->args([
+                tagged_locator('controller.argument_value_resolver', 'name'),
+            ])
+            ->tag('controller.argument_value_resolver', ['priority' => 1000])
+
         ->set('argument_resolver.backed_enum_resolver', BackedEnumValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => 100])
+            ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => 'backed_enum'])
 
         ->set('argument_resolver.uid', UidValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => 100])
+            ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => 'uid'])
 
         ->set('argument_resolver.datetime', DateTimeValueResolver::class)
             ->args([
                 service('clock')->nullOnInvalid(),
             ])
-            ->tag('controller.argument_value_resolver', ['priority' => 100])
+            ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => 'datetime'])
 
         ->set('argument_resolver.request_attribute', RequestAttributeValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => 100])
+            ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => 'request_attribute'])
 
         ->set('argument_resolver.request', RequestValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => 50])
+            ->tag('controller.argument_value_resolver', ['priority' => 50, 'name' => 'request'])
 
         ->set('argument_resolver.session', SessionValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => 50])
+            ->tag('controller.argument_value_resolver', ['priority' => 50, 'name' => 'session'])
 
         ->set('argument_resolver.service', ServiceValueResolver::class)
             ->args([
                 abstract_arg('service locator, set in RegisterControllerArgumentLocatorsPass'),
             ])
-            ->tag('controller.argument_value_resolver', ['priority' => -50])
+            ->tag('controller.argument_value_resolver', ['priority' => -50, 'name' => 'service'])
 
         ->set('argument_resolver.default', DefaultValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => -100])
+            ->tag('controller.argument_value_resolver', ['priority' => -100, 'name' => 'default'])
 
         ->set('argument_resolver.variadic', VariadicValueResolver::class)
-            ->tag('controller.argument_value_resolver', ['priority' => -150])
+            ->tag('controller.argument_value_resolver', ['priority' => -150, 'name' => 'variadic'])
```

Comment:
what about using the FQCN of the service class instead of inventing a nickname?

Reply:
This would be simpler but will prevent usage of resolvers if they are many using the same class. Still not sure what should be the stance on this matter.

Reply:
that wouldn't: same-class resolver would be given another name, that's all

Reply:
Ah yes so we would only require this attribute to be present in such case right?

Reply:
we should always define it, or the default name will be the service id, which is not good
We should use `'name' => VariadicValueResolver::class` on this line.

Reply:
Oh okay I thought a compiler pass could create the service locator and set the appropriate identifiers.

That means every resolver definition will have to be added this attribute.

Reply:
Because we register services in app/ as FQCN, there should be nothing to do for them. For bundles, yes of course.

Reply:
Okay so PRs on the SecurityBundle, DoctrineBundle and PSR-7 Bridge will follow :ok_hand: